integration_tests/tests/assert_fetch_objects_equal.sql

```diff
@@ -0,0 +1,19 @@
+-- depends_on: {{ ref('data_fetch') }}
+
+{% set expected_dictionary={
+    'col_1': (1, 2, 3),
+    'col_2': ('a', 'b', 'c'),
+    'col_3': (4.0, 5.0, none)
+} %}
+
+{% set actual_dictionary=dbt_utils.fetch(
+    "select * from {{ ref('data_fetch') }}"
```

Comment:
same thing here - I don't think you want to use curly brackets inside of a `set` definition.  Try:
```
"select * from " ~ ref('data_fetch')
```